docs/utilities/parser.md

```diff
@@ -249,3 +249,51 @@ The package `@types/aws-lambda` is a popular project that contains type definiti
 Powertools parser utility also bring AWS Lambda event types based on the built-in schema definitions.
 
 We recommend to use the types provided by the parser utility. If you encounter any issues or have any feedback, please [submit an issue](https://github.com/aws-powertools/powertools-lambda-typescript/issues/new/choose).
+
+## Testing your code
+
+When testing your handler with **parser decorator** you need to use double assetion to bypass TypeScript type checking in your tests.
+This is useful when you want to test the handler for invalid payloads or when you want to test the error handling.
+If you are you use middy middleware, you don't need to do this.
```

Comment:
Just for my understanding - aka not challenging the statement - why does this not apply to the middleware?

Reply:
I can see that `middy()` return type is ` MiddyfiedHandler` which accepts any event structure with `TEvent=any`. Our parser middleware implementation can not pass the type information which was raised in #2407 , so this might change soon.